spec/models/piece_spec.rb

```diff
@@ -1,18 +1,232 @@
+        @white_piece = FactoryGirl.create(:piece, file: 1, rank: 1, color: 'white', game: @game)
+        empty_position_file = 2
+        empty_position_rank = 2
+
+        result = @white_piece.is_capture_opposing_color?(empty_position_file, empty_position_rank)
+
+        expect(result).to be false
+      end
+    end
+
+  end
+
+  describe "piece#is_unidirectional_type?" do
+    before do
+      @game = FactoryGirl.create(:game)
+    end
+
+    it "is unidirectional" do 
+      @piece = FactoryGirl.create(:piece)
+      unidirectional_types = ["Rook", "Queen", "Bishop"]
+
+      unidirectional_types.each do |chess_type|
+        @piece.type = chess_type
+
+        result = @piece.is_unidirectional_type?
+
+        expect(result).to be true
+      end
+    end
+
+    it "is not unidirectional" do
+      @piece = FactoryGirl.create(:piece)
+      unidirectional_types = ["Pawn", "King", "Knight"]
+
+      unidirectional_types.each do |chess_type|
+        @piece.type = chess_type
+
+        result = @piece.is_unidirectional_type?
+
+        expect(result).to be false
+      end
+    end
+  end
+
+  describe ".valid_move? validates piece move positions" do
+
+    before do
+      @game = FactoryGirl.create(:game)
+    end
+
+    context "valid move" do
+      it "when moving inside the board" do
+        @piece = FactoryGirl.create(:piece, file: 1, rank: 1, game: @game)
+        new_file = 2
+        new_rank = 2
+
+        result = @piece.valid_move?(new_file, new_rank)
+
+        expect(result).to be true
+      end
+
+      it "when moving to an opposing piece position" do
+        @white_piece = FactoryGirl.create(:piece, file: 1, rank: 1, color: 'white', game: @game)
+        @black_piece = FactoryGirl.create(:piece, file: 2, rank: 2, color: 'black', game: @game)
+
+        result = @white_piece.valid_move?(@black_piece.file, @black_piece.rank)
+
+        expect(result).to be true
+      end
+    end
+
+    context "invalid move" do
+      it "when moving off the board" do
+        @piece = FactoryGirl.create(:piece, file: 1, rank: 1, game: @game)
+        invalid_file = -1
+        invalid_rank = -1
+
+        result = @piece.valid_move?(invalid_file, invalid_rank)
+
+        expect(result).to be false
+      end
+
+      it "when moving to a same color piece position" do
+        @white_piece = FactoryGirl.create(:piece, file: 1, rank: 1, color: 'white', game: @game)
+        @same_color_piece = FactoryGirl.create(:piece, file: 2, rank: 2, color: 'white', game: @game)
+        
+        result = @white_piece.valid_move?(@same_color_piece.file, @same_color_piece.rank)
+
+        expect(result).to be false
+      end
+
+      it "when moving in place" do
+        @piece = FactoryGirl.create(:piece, file: 1, rank: 1, game: @game)
+
+        result = @piece.valid_move?(@piece.file, @piece.rank)
+
+        expect(result).to be false
+      end
+
+      context "pieces that move more than 1 space unidirectionally" do
+
```

Comment:
I have trouble understanding this :grimacing: :snail:
Some thoughts: we should make a change to the `is_obstructed?` method to include the special case of the Knight never getting obstructed.  Regardless of this PR I think `.is_obstructed?` needs to be amended this way and we could do this separately.  This way `.is_unidirectional_type?` may not be necessary??

I suggest we add `return false if type == "Knight"` at the first line under `def is_obstructed?(col, row)` in piece model. We could test it together with `piece#is_obstructed?` unit tests; something like:

```
it "Knight never is obstructed" do
    knight = FactoryGirl.create(:piece, type: "Knight", file: 2, rank: 1, game: @game)
    FactoryGirl.create(:piece, file: 2, rank: 2, game: @game)
    expect(knight.is_obstructed?(1, 3)).to be false
end
```
Then we could pull `.is_obstructed?` in `.valid_move?`

Reply:
Yeah....I had a feeling this might've been confusing 😖  since I had a hard time thinking of what to name this method.  I like your suggestion of pulling the knight type edge case into is_obstructed? make things more concise.